karton/classifier/classifier.py

```diff
@@ -111,33 +129,37 @@ def _get_extension(self, name: str) -> str:
         splitted = name.rsplit(".", 1)
         return splitted[-1].lower() if len(splitted) > 1 else ""
 
-    def _classify(self, task: Task) -> Optional[Dict[str, str]]:
+    def _classify(self, task: Task) -> Optional[Dict[str, Optional[str]]]:
         sample = task.get_resource("sample")
         content = cast(bytes, sample.content)
 
         magic = task.get_payload("magic") or ""
+        magic_mime = task.get_payload("mime") or ""
         try:
             magic = pymagic.from_buffer(content)
+            magic_mime = pymagic.from_buffer(content, mime=True)
         except Exception as ex:
             self.log.warning(f"unable to get magic: {ex}")
 
         extension = self._get_extension(sample.name or "sample")
-        sample_type = {
-            "type": "sample",
-            "stage": "recognized",
-            "quality": task.headers.get("quality", "high"),
+        sample_class = {
+            "magic": magic if magic else None,
+            "mime": magic_mime if magic_mime else None,
+            "kind": None,
+            "platform": None,
+            "extension": None,
```

Comment:
I think it would be cleaner to return a dict with non-nullable values. We would avoid typing changes and use existing pattern `if key in dict: ...` instead of rewriting it to `.get(key) is not None`

Reply:
Do not agree with your comment. Knowing that a certain key is always present makes it more readable, since you know what to expect. Moreover, this way you see which values are to be "filled up" in this long method.